src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -104,91 +109,145 @@ private static Task StartProcessWithKeepAlive(ProcessMetadata metadata)
+                                    // This means we have tried to start from this domain before
+                                    metadata.ProcessState = metadata.ProcessIsHealthy() ? ProcessState.Healthy : ProcessState.ReadyToStart;
                                 }
 
-                                Log.Debug("Starting {Process}.", path);
-                                metadata.Process = Process.Start(startInfo);
-
-                                await Task.Delay(150, _cancellationTokenSource.Token).ConfigureAwait(false);
-
-                                if (metadata.Process == null || metadata.Process.HasExited)
-                                {
-                                    Log.Error("{Process} has failed to start.", path);
-                                    sequentialFailures++;
-                                }
-                                else
+                                if (metadata.ProcessState == ProcessState.ReadyToStart)
                                 {
-                                    Log.Debug("Successfully started {Process}.", path);
-                                    sequentialFailures = 0;
+                                    Log.Information("Attempting to start {Process}.", path);
+
+                                    var startInfo = new ProcessStartInfo
+                                    {
+                                        FileName = path,
+                                        UseShellExecute = false // Force consistency in behavior between Framework and Core
+                                    };
+
+                                    if (!string.IsNullOrWhiteSpace(metadata.ProcessArguments))
+                                    {
+                                        startInfo.Arguments = metadata.ProcessArguments;
+                                    }
+
+                                    metadata.Process = Process.Start(startInfo);
+                                    var timeout = 2000;
+
+                                    while (timeout > 0)
+                                    {
+                                        // Loop and watch for evidence that the process is up and running
+                                        if (metadata.ProcessIsHealthy())
+                                        {
+                                            metadata.SequentialFailures = 0;
+                                            metadata.ProcessState = ProcessState.Healthy;
+                                            Log.Information("Successfully started {Process}.", path);
+                                            break;
+                                        }
+
+                                        if (metadata.Process == null || metadata.Process.HasExited)
+                                        {
+                                            Log.Error("{Process} has failed to start.", path);
```

Comment:
Small tweak for consistency with the `Successfully started {Process}` message above.
```suggestion
                                            Log.Error("Failed to start {Process}.", path);
```